plugins/modules/source_control/gitlab/gitlab_user.py

```diff
@@ -219,10 +250,13 @@ def createOrUpdateUser(self, username, options):
                 'name': options['name'],
                 'username': username,
                 'password': options['password'],
+                'reset_password': options['reset_password'],
                 'email': options['email'],
                 'skip_confirmation': not options['confirm'],
                 'admin': options['isadmin'],
-                'external': options['external']})
+                'external': options['external'],
+                'provider': options['provider'],
+                'extern_uid': options['extern_uid']})
```

Comment:
This way, the next diff will be nicer.
```suggestion
                'extern_uid': options['extern_uid'],
            })
```